EngineLayer/DecoyOnTheFly.cs

```diff
@@ -0,0 +1,168 @@
+﻿using Proteomics;
+using Proteomics.ProteolyticDigestion;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace EngineLayer
+{
+    public class DecoyOnTheFly
+    {
+        //This function maintains the amino acids associated with the protease motif and reverses all other amino acids.
+        //N-terminal modificatons are preserved. Other modifications travel with their respective amino acids. this results
+        //in a decoy peptide composed the same amino acids and modifications as the original. 
+        //Occasionally, this process results in peptide with exactly the same sequence. Therefore, there is a stop-gap measure
+        //the returns the mirror image of the original. N-terminal mods are preserved, but other mods are also reversed. 
+        //this should yield a unique decoy for each target sequence.
+        public static PeptideWithSetModifications GetReverseDecoyFromTarget(PeptideWithSetModifications pwsm, int[] revisedAminoAcidOrder)
```

Comment:
I feel like this function should be in mzlib, rather than MetaMorpheus. We currently have all of our digestion/decoy code in mzLib, so there's a precedent.